components/tidb_query_expr/src/impl_string.rs

```diff
@@ -679,9 +679,17 @@ pub fn substring_index(
 
 #[rpn_fn]
 #[inline]
-pub fn strcmp(left: BytesRef, right: BytesRef) -> Result<Option<i64>> {
+pub fn strcmp<C: Collator>(left: BytesRef, right: BytesRef) -> Result<Option<i64>> {
     use std::cmp::Ordering::*;
-    Ok(Some(match left.cmp(right) {
+    let left = match String::from_utf8(left.to_vec()) {
```

Comment:
Convert to `String` requires extra copy, try convert to `str` like: `str::from_utf8(left)`